File: Scripts/Python/Aug_27/legendre.py
#!/usr/bin/env python
# File: legendre.py
# Name: D.Saravanan
# Date: 27/08/2021

""" Script to calculate and plot Legendre polynomials """

import numpy as np
import matplotlib
matplotlib.rcParams['pgf.texsystem'] = 'pdflatex'
matplotlib.rcParams.update({'font.family': 'serif', 'font.size': 8,
    'axes.labelsize': 10, 'axes.titlesize': 10, 'figure.titlesize': 10})
matplotlib.rcParams['text.usetex'] = True
import matplotlib.pyplot as plt
matplotlib.use('Agg')

fig, ax = plt.subplots()

for n in range(5):

    # generate the Legendre basis for the polynomial of degree 'n'
    Pn = np.polynomial.legendre.Legendre.basis(n)

    # interval [-1, 1]
    x = np.linspace(-1, 1, 1000) 
    y = Pn(x)

    #x, y = np.polynomial.Legendre.basis(n, [-1,1]).linspace(1000)
    #x, y = np.polynomial.legendre.Legendre.basis(n, [-1,1]).linspace(1000)

    ax.plot(x, y, label=r'$P_{}(x)$'.format(n))

ax.set(xlabel=r'$x$', ylabel=r'$P_{n}(x)$')
ax.set_title(r'Legendre polynomials')
ax.grid(); ax.legend()
plt.savefig('legendre.pdf')

File: Scripts/Julia/Aug_27/legendre.jl
#!/usr/bin/env julia
# File: legendre.jl
# Name: D.Saravanan
# Date: 27/08/2021

""" Script to calculate and plot Legendre polynomials """

import PyPlot
const plt = PyPlot
plt.rc("pgf", texsystem="pdflatex")
plt.rc("font", family="serif", weight="normal", size=8)
plt.rc("axes", labelsize=10, titlesize=10)
plt.rc("figure", titlesize=10)
plt.rc("text", usetex="True")
using LaTeXStrings

function Legendre(x, n)
    # recurrence relation of Legendre polynomials
    # P_0 = 1, P_1 = x
    # (n+1)  P_(n+1) = (2n+1) x P_(n) - n P_(n-1)
    n == 0 && return 1
    n == 1 && return x
    ((2(n-1) + 1)*x*Legendre(x, n-1) - (n-1)*Legendre(x, n-2))/n
end

fig, ax = plt.subplots()

for n in 0:4
    
    # interval [-1, 1]
    x = LinRange(-1, 1, 1000)
    y = map(x -> Legendre(x, n), x)

    ax.plot(x, y, label=L"P_{n}(x)")
end

ax.set(xlabel=raw"$x$", ylabel=raw"$P_{n}(x)$")
ax.set_title(raw"Legendre polynomials")
ax.grid(true); ax.legend()
plt.savefig("legendre.png")
